cpp/modmesh/linalg/factorization.hpp

```diff
@@ -28,7 +28,7 @@
  * POSSIBILITY OF SUCH DAMAGE.
  */
 
-#include <modmesh/buffer/SimpleArray.hpp>
+#include <modmesh/buffer/buffer.hpp>
```

Comment:
When being outside of the directory (as a natural separation of a module in modmesh C++ code), include the master include file, not the individual include file.

FYI @ThreeMonth03